### PR TITLE
Add fractional-delay polyphase FIR helpers

### PR DESCRIPTION
**Pull request by xerpi**
Add helper-only fractional-delay support on top of `PolyphaseFir`. The new
`polyphase::fractional_delay` module documents the phase `i / M` geometry, clarifies that each
branch is a full input-rate FIR, and provides dense prototype length helpers.

Add a shared `filter_bank::pack_prototype_taps` reorderer for phase-major coefficient storage, and
use it from `PolyphaseFilterBankVec::from_prototype_taps`. Add Kaiser low-pass prototype helpers for
caller-provided storage plus a `Vec` convenience.

`PolyphaseFir::reset_delay_line` and the generic `zero_fill_ring` helper let stateful users reset
reused delay-line storage without rebuilding coefficients.

### CODE REVIEW

File: src/filters/fir/polyphase/fractional_delay.rs [hunk 1]
```diff
@@ -0,0 +1,947 @@
...
+    order: KaiserOrder<T>,
+    cutoff: T,
+) where
+    T: Float + core::fmt::Debug,
+{
+    assert!(num_phases > 0, "fractional-delay bank needs phase branches");
+    assert!(
+        order.num_taps > 0,
+        "fractional-delay bank taps per phase must be nonzero"
+    );
```
Comment (regexident) [inline src/filters/fir/polyphase/fractional_delay.rs:270]: The check allows `num_taps == 1`, but downstream `kaiser::lowpass_with_beta` requires `N >= 2` and panics with a confusing module-level error about windowed-sinc being degenerate. A manually constructed `KaiserOrder { num_taps: 1, .. }` passes this public API guard but fails one call deeper with an unrelated message. While `kaiser_order` never returns `num_taps == 1`, this is a defensive precondition gap at the public API boundary.

File: src/filters/fir/polyphase/fractional_delay.rs [hunk 2]
```diff
@@ -0,0 +1,947 @@
...
+/// before packing so every branch has approximately unit DC gain after
+/// polyphase decomposition and preserves constant inputs. Phase 0 matches the
+/// ordinary sample-rate Kaiser low-pass defined by `order` and `cutoff`; later
+/// phases are fractional shifts of the same filter.
+///
+/// # Panics
+///
+/// Panics when `num_phases` or `order.num_taps` is zero, when `prototype.len()`
+/// does not equal `prototype_len(order.num_taps, num_phases)`, when `cutoff` is
+/// outside `(0, 0.5)`, or when `num_phases` cannot be represented as `T`.
```
Comment (regexident) [inline src/filters/fir/polyphase/fractional_delay.rs]: Could we add `#[should_panic]` tests for `kaiser_lowpass_prototype` with `num_phases = 0` and `order.num_taps = 0`?

Doc lists panics on zero `num_phases`, zero `order.num_taps`, and non-representable `num_phases`. The test module tests `prototype_len` zero-input panics and length-mismatch/above-Nyquist panics, but not these three documented panic paths. Two are straightforward boundary conditions; the third is harder to trigger but still listed in documentation.

File: src/filters/fir/polyphase/fractional_delay.rs [hunk 3]
```diff
@@ -0,0 +1,947 @@
...
+//!    zero-pads the tail.
+//! 5. Feed samples to [`PolyphaseFir::push`](super::fir::PolyphaseFir::push) and
+//!    read a branch with
+//!    [`PolyphaseFir::execute`](super::fir::PolyphaseFir::execute).
+//!
+//! Steps 3 and 4 can share one buffer instead of two. Allocate the packed
+//! length, generate the dense kernel into `taps[..dense_len]`, then call
+//! [`pack_prototype_taps_in_place`](super::filter_bank::pack_prototype_taps_in_place)
+//! over the whole slice. The subslice is required because the design helpers
+//! assert an exact prototype length.
```
Comment (regexident) [inline src/filters/fir/polyphase/fractional_delay.rs]:
```suggestion
//! Steps 3 and 4 can share one buffer instead of two. Allocate the packed
//! length, generate the dense kernel into `taps[..dense_len]`, then call
//! [`pack_prototype_taps_in_place`](super::filter_bank::pack_prototype_taps_in_place)
//! over the whole slice. The subslice is required because the design helpers
//! assert an exact prototype length.
//!
//! The in-place path uses a rotation-based transpose with higher constant-factor cost;
//! when initialization time matters, use the two-buffer route instead.
```

Docs recommend the in-place single-buffer path as a space optimization without noting time cost. `matrix_transpose_in_place` is O(W²H²/4) element moves. For a 128-phase, 32-tap-per-phase bank, this is ~4M moves for 4K elements: ~1000× overhead vs the O(n) two-buffer route.

The complexity is documented on `matrix_transpose_in_place` but not propagated to the user-facing module level.

**Review by regexident — COMMENTED** [review]
Thanks @xerpi, just some minor nits.

File: src/filters/fir/polyphase/fractional_delay.rs (hunk 1 above)
Comment (xerpi) [reply src/filters/fir/polyphase/fractional_delay.rs:270]: Good point, I'll tighten it.

File: src/filters/fir/polyphase/fractional_delay.rs (hunk 2 above)
Comment (xerpi) [reply src/filters/fir/polyphase/fractional_delay.rs]: Will add the first two.
Regarding the third path, non-representable `num_phases`: I don't think it's reachable for `f32` or `f64`. `NumCast::from(usize::MAX)` returns `Some(1.84e19)`, which is lossy, but never `None`. So the `T::from(num_phases)` guard can't be tripped with the float types in `std`.

File: src/filters/fir/polyphase/fractional_delay.rs (hunk 3 above)
Comment (xerpi) [reply src/filters/fir/polyphase/fractional_delay.rs]: Good point, will add a `Complexity` section to `pack_prototype_taps_in_place` and will make `fractional_delay.rs` module docs refer to it.
